aioauth/server.py

```diff
@@ -268,9 +268,8 @@ async def token(request: fastapi.Request) -> fastapi.Response:
             # client_secret is required for the client_credentials grant type
             # https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/
             client_id, client_secret = self.get_client_credentials(request)
-        elif request.post.grant_type == "password":
-            # client_secret is optional for the password grant type
-            # https://www.oauth.com/oauth2-servers/access-tokens/password-grant/
+        else:
+            # for other grant types, client_secret is required if the client has one.
```

Comment:
@tdg5 Could you also add this piece of documentation here? "If the client type is confidential or the client was issued client credentials (or assigned other authentication requirements), the client MUST authenticate with the authorization server as described in Section 3.2.1." as a reminder of the rationale behind this piece of code.

Reply:
let me know if I can merge this.